README.md

```diff
@@ -12,62 +12,361 @@ Install from NPM:
 ```
 
-## Developing, Testing and Contributing
+## Writing Tests
 
-After cloning the repo, run `npm install` to install dependencies.
+Tests leverage the familiar [Mocha](https://mochajs.org/) and [Chai](https://www.chaijs.com/) test frameworks. Many of the available utilities are wrappers around or extensions of the excellent [@open-wc testing helpers](https://open-wc.org/docs/testing/helpers/).
 
-### Linting
+### describe, it, assert & expect
 
-```shell
-# eslint
-npm run lint
+Tests can be grouped into suites using `describe` and are defined inside `it` blocks.
+
+Results can be verified using either the BDD-style `expect` or TDD-style `assert` syntax (although try not to mix & match).
+
+```javascript
+import { assert, expect } from '@brightspace-ui/testing';
+
+describe('group of tests', () => {
+  it('should multiply numbers', () => {
+    expect(2 * 4).to.equal(8);
+  });
+  it('should subtract numbers', () => {
+    assert.equal(4 - 2, 2);
+  });
+});
 ```
 
-### Testing
+### Testing UI with Fixtures
 
-```shell
-# lint & run headless unit tests
-npm test
+To run tests against snippets of HTML (including web components), use the `html` string literal and the asynchronous `fixture()`.
+
+```javascript
+import { expect, html, fixture } from '@brightspace-ui/testing';
+
+it('should have the correct class', async() => {
+  const elem = await fixture(html`<div class="foo"></div>`);
+  expect(elem).classList.contains('foo').to.be.true;
+});
+```
+
+Each call to `fixture()` will restore the browser to its default state, making subsequent calls isolated from each other. There's therefore no need to manually restore the viewport, language, mouse, or keyboard in between tests.
+
+#### Configuring the Viewport Size
+
+The viewport defaults to `800px` wide by `800px` tall. To use different viewport sizes, pass a `viewport` option to `fixture()`.
+
+```javascript
+it('should work on small viewports', async() => {
+  const elem = await fixture(html`<my-elem></my-elem>`, {
+    viewport: { height: 300, width: 200 }
+  });
+  // do assertions
+});
+```
+
+#### Configuring the Language or Text Direction
+
+If the component under test has special multi-lingual or bidirectional text behavior, both `language` and `rtl` (right-to-left) options are available.
+
+```javascript
+it('should work in French', async() => {
+  const elem = await fixture(html`<my-elem></my-elem>`, {
+    lang: 'fr'
+  });
+  // do assertions
+});
+
+it('should work in RTL', async() => {
+  const elem = await fixture(html`<my-elem></my-elem>`, {
+    rtl: true
+  });
+  // do assertions
+});
+```
+
+If `lang` is set to Arabic (`ar`), the right-to-left option will automatically be enabled.
+
+> **Note:** it's not recommended to use `language` configuration with visual-diff to solely test the correctness of translations. The [messageformat-validator](https://github.com/bearfriend/messageformat-validator) is a more efficient way to test translations.
```

Comment:
Lol we should probably get that moved into `BrightspaceUI` at some point

Reply:
Yeah, or even into this library! `d2l-test-mfv`